src/utils/devices.js

```diff
@@ -0,0 +1,3 @@
+/**
+ * @typedef {'cpu'|'gpu'|'wasm'|'webgpu'|null} DeviceType
```

Comment:
It's meant to be a "catch-all" for the different ways that the library can be used with GPU support (not just in the browser with WebGPU). The idea is that it will simplify documentation, as transformers.js will select the best execution provider depending on the environment. For example, DML/CUDA support in onnxruntime-node (see https://github.com/microsoft/onnxruntime/pull/16050#issuecomment-1909421315)

Of course, this is still a work in progress, so it can definitely change!